Doc updates for compression  on continuous aggregate feature (#666)
timescale/docs: timescaledb/how-to-guides/continuous-aggregates/compression-on-continuous-aggregates.md

continuous aggregate. To do this, you have to manually decompress 
the compressed chunk and then execute a `refresh_continuous_aggregate` call.

## Enable compression on continuous aggregates
You can enable and disable compression on continuous aggregated by setting
`compress` parameter when you alter the view.

<procedure>

### Enabling and disabling compression on continuous aggregates
1.  For an existing continuous aggregate, at the `psql` prompt, enable
 compression:
    ```sql
    ALTER MATERIALIZED VIEW cagg_name set (timescaledb.compress = true);
    ```
1.  Disable compression:
    ```sql
    ALTER MATERIALIZED VIEW cagg_name set (timescaledb.compress = false);
    ```
</procedure>
The decompress command fails if there are compressed chunks associated with the 
continuous aggregate. In this case, you need to decompress the chunks, and then 
drop any compression policy on the continuous aggregate, before you disable 
compression. For more detailed information, see the
[decompress chunks] [decompress-chunks] section:
```sql
SELECT decompress_chunk(c, true) FROM show_chunks('cagg_name') c;
 

## Compression policies on continuous aggregates

> Before  setting up a compression policy on a continuous aggregate, you should

Reviewer: should we specify that you _have_ to set up a refresh policy before you set up a compression policy?

Reviewer 2: "must" ?

Reviewer 3: Currently, you really "have to" based on yesterday's conversation.

# Compression on continuous aggregates

> Continuous aggregates are often used to store down sampled historical data.

Reviewer 2: ```suggestion
Continuous aggregates are often used to store downsampled historical data.
```

The historical data is almost never modified or recomputed and is only used 
for serving analytic queries. For this use case, it is often beneficial to 
store the materialized data in compressed form to save on storage costs. 
You can get these cost savings by enabling compression on continuous 
aggregates.

> At this time, TimescaleDB does not support refreshing compressed regions of a 

Reviewer 2: ```suggestion
Currently, TimescaleDB does not support refreshing compressed regions of a 
```

> continuous aggregate. In order to do this, you will have to manually decompress 

Reviewer 2: ```suggestion
continuous aggregate. To do this, you have to manually decompress 
```

the compressed chunk and then execute a `refresh_continuous_aggregate` call.

## Enable compression on continuous aggregates
You can enable and disable compression on continuous aggregated by setting
`compress` parameter when you alter the view.

<procedure>

> ### Enabling/Disabling compression on continuous aggregate aggregation

Reviewer 2: ```suggestion
### Enabling and disabling compression on continuous aggregates
```

1.  For an existing continuous aggregate, at the `psql` prompt, enable
 compression:

> 

Reviewer 2: ```suggestion
```

    ```sql
    ALTER MATERIALIZED VIEW cagg_name set (timescaledb.compress = true);
    ```

> 1.  Disable compression. 
> 

Reviewer 2: ```suggestion
1.  Disable compression:
```

    ```sql
    ALTER MATERIALIZED VIEW cagg_name set (timescaledb.compress = false);
    ```

>     

Reviewer 2: ```suggestion
</procedure>
```

>     This command fails if there are compressed chunks associated with the 
>     continuous aggregate. If there are compressed chunks, first
>     decompress the chunks and then drop any compression policy on the continuous
>     aggregate, before trying to disable compression. For
>     more detailed information, see [Decompress chunks] [decompress-chunks]
>   
>     ```sql
>     SELECT decompress_chunk(c, true) FROM show_chunks('cagg_name') c;
>     ```

Reviewer 2: ```suggestion
The decompress command fails if there are compressed chunks associated with the 
continuous aggregate. In this case, you need to decompress the chunks, and then 
drop any compression policy on the continuous aggregate, before you disable 
compression. For more detailed information, see the
[decompress chunks] [decompress-chunks] section:
```sql
SELECT decompress_chunk(c, true) FROM show_chunks('cagg_name') c;
```
```

> </procedure>

Reviewer 2: ```suggestion
```

## Compression policies on continuous aggregates
Before  setting up a compression policy on a continuous aggregate, you should
set up a refresh policy. The compression policy interval should be set so that
actively refreshed regions are not compressed. This is to prevent refresh

> policies from failing.  To illustrate, with a refresh policy like so:
> 

Reviewer 2: ```suggestion
policies from failing. For example, consider a refresh policy like this:

```

```sql
SELECT add_continuous_aggregate_policy('cagg_name', refresh_start=>'30 days', refresh_end=>'1 day', '1 h');
```

> The compression policy should be set so that `compress_after` parameter is greater than the `refresh_start` parameter of the continuous aggregate policy.
> 

Reviewer 2: ```suggestion
With this kind of refresh policy, the compression policy needs the `compress_after` 
parameter greater than the `refresh_start` parameter of the continuous aggregate policy:
```

Author: Its not specific to this example. 
compress_after should always be greater than refresh_start. 
The example is intended to illustrate how to go about doing that.

Reviewer 2: ```suggestion
We recommend that the `compress_after` parameter is greater than the `refresh_start` parameter of the continuous aggregate policy:

```

Author: Well, it throws an error if it does not meet the condition. That's the reason I phrased it as should.

Author: ```
test=# select add_compression_policy('metrics_cagg', ' 1 h'::interval);
ERROR:  compress_after value for compression policy should be greater than the start of the refresh window of continuous aggregate policy for metrics_cagg
```

Reviewer 2: ```suggestion
The `compress_after` parameter must be greater than the `refresh_start` parameter of the continuous aggregate policy:

```

```sql
SELECT add_compression_policy('cagg_name', compress_after=>'45 days'::interval);
```

> Once a chunk is compressed, manual refresh calls that attempt to refresh the 
> compressed region will fail with an error as shown below.
> 

Reviewer 2: ```suggestion
When a chunk is compressed, manual refresh calls that attempt to refresh the 
compressed region fail with an error like this:
```

Author: Repharsed to add more clarity.
